components/Form.js

```diff
@@ -14,78 +25,101 @@ export default function CreateForm({ onCreateItem }) {
     const formData = new FormData(event.target);
     const newItem = Object.fromEntries(formData);
 
-    onCreateItem(newItem);
-    event.target.reset();
-    setSelectedCategory("");
+    if (editItem) {
+      onEditItem(editItem.id, newItem);
+    } else {
+      onCreateItem(newItem);
+    }
+
+    onToggleIsEditing(null);
   }
 
   return (
-    <StyledForm onSubmit={handleSubmit}>
-      <StyledHeader>Add new item</StyledHeader>
-      <label htmlFor="name">Item Name</label>
-      <StyledItemNameInput
-        name="name"
-        id="name"
-        type="text"
-        placeholder="(Required)"
-        maxLength="20"
-        required
-      />
-      <FlexContainer>
-        <StyledCategorySelectContainer>
-          <label htmlFor="category">Category</label>
-          <StyledCategorySelect
-            name="category"
-            id="category"
-            value={selectedCategory}
-            onChange={handleCategoryChange}
-            required
-          >
-            <option value="">---Choose a category---</option>
-            {categories.map((category) => {
-              return (
-                <option key={category} value={category}>
-                  {category}
-                </option>
-              );
-            })}
-          </StyledCategorySelect>
-        </StyledCategorySelectContainer>
-
-        <StyledQuantityInputContainer>
-          <label htmlFor="quantity">Quantity</label>
-          <StyledQuantityInput
-            name="quantity"
-            id="quantity"
-            type="number"
-            placeholder="(Required)"
-            min="0"
-            max="999"
-            required
-          />
-        </StyledQuantityInputContainer>
-      </FlexContainer>
-      <label htmlFor="comment">Comment</label>
-      <StyledCommentTextarea
-        name="comment"
-        id="comment"
-        placeholder="(Optional)"
-        maxLength="100"
-        wrap="hard"
-      ></StyledCommentTextarea>
-      <StyledCreateButton>Create</StyledCreateButton>
-    </StyledForm>
+    <FormContainer>
+      <StyledForm onSubmit={handleSubmit}>
+        {!editItem && <StyledHeader>Add new item</StyledHeader>}
+        {editItem && <StyledHeader>Edit shopping item</StyledHeader>}
```

Comment:
Instead of repeating the StyledHeader component twice, you can simplify the conditional rendering by using a **ternary operator**. But if your way is easier for you to understand, then leave it as it is.  

```suggestion
      <StyledHeader>{editItem ? "Edit shopping item" : "Add new item"}</StyledHeader>
```

What does the ternary do:
- The ternary operator `editItem ? "Edit shopping item" : "Add new item"` checks if editItem exists.
- If `editItem` is truthy (i.e., you're editing an item), it will display "Edit shopping item".
- Otherwise, it will display "Add new item" (for creating a new item).